### PR TITLE
[MINOR][SQL] Remove unused pattern matching variables in Optimizers.

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This PR removes unused pattern matching variable in Optimizers in order to improve readability.
## How was this patch tested?

Pass the existing Jenkins tests.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala [hunk 1]
```diff
@@ -440,10 +439,10 @@ object ColumnPruning extends Rule[LogicalPlan] {
     case w: Window if w.windowExpressions.isEmpty => w.child
 
     // Eliminate no-op Projects
-    case p @ Project(projectList, child) if sameOutput(child.output, p.output) => child
+    case p @ Project(_, child) if sameOutput(child.output, p.output) => child
 
     // Can't prune the columns on LeafNode
-    case p @ Project(_, l: LeafNode) => p
+    case p @ Project(_, _: LeafNode) => p
```
Comment (srowen) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala]: in this case, just `p: Project =>`?

**Comment by srowen** [comment]
Looks reasonable, but I wonder if this should be done at once for all of the code base?

**Comment by dongjoon-hyun** [comment]
Thank you for review, @srowen .
Probably, there exists similar cases in other modules, too.

But, for the other module, I guess the patterns are simple and the most complex pattern matchings are used in optimizers due to the nature of optimization. Also, until now, it seems there exists no suitable automatic tool  for doing for all of the code base. When I searched for this, I found that Scala open issue for this; https://issues.scala-lang.org/browse/SI-8040 .

**Comment by srowen** [comment]
Merged to master

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen.

**Comment by srowen** [comment]
Also merged to 2.0 -- seems safe enough and might avoid merge conflicts

**Pull request closed without merging**